src/graph.rs

```diff
@@ -725,5 +762,5 @@ fn serialize_source<S>(source: &str, serializer: S) -> Result<S::Ok, S::Error>
 where
   S: Serializer,
 {
-  serializer.serialize_u32(source.as_bytes().iter().count() as u32)
```

Comment:
Does this iterate over all the bytes and get a count or is it somehow able to use [`ExactSizeIterator`](https://doc.rust-lang.org/std/iter/trait.ExactSizeIterator.html) to figure out the length? I wasn't sure so I switched it to use `.len()` to get the byte count which I think is more standard and maybe faster.

Reply:
Yeah, I think it was a hold-over from the logic in CLI, and I didn't question it.  It looks like `.len()` is as _valid_ as the more obtuse.